internal/service/codepipeline/codepipeline.go

```diff
@@ -400,213 +610,8 @@ func resourcePipeline() *schema.Resource {
-																		ValidateFunc: validation.StringLenBetween(1, 255),
-																	},
-																},
-															},
-														},
-													},
-													"file_paths": {
-														Type:     schema.TypeList,
-														Optional: true,
-														MaxItems: 1,
-														Elem: &schema.Resource{
-															Schema: map[string]*schema.Schema{
-																"excludes": {
-																	Type:     schema.TypeList,
-																	Optional: true,
-																	MinItems: 1,
-																	MaxItems: 8,
-																	Elem: &schema.Schema{
-																		Type:         schema.TypeString,
-																		ValidateFunc: validation.StringLenBetween(1, 255),
-																	},
-																},
-																"includes": {
-																	Type:     schema.TypeList,
-																	Optional: true,
-																	MinItems: 1,
-																	MaxItems: 8,
-																	Elem: &schema.Schema{
-																		Type:         schema.TypeString,
-																		ValidateFunc: validation.StringLenBetween(1, 255),
-																	},
-																},
-															},
-														},
-													},
-													names.AttrTags: {
-														Type:     schema.TypeList,
-														Optional: true,
-														MaxItems: 1,
-														Elem: &schema.Resource{
-															Schema: map[string]*schema.Schema{
-																"excludes": {
-																	Type:     schema.TypeList,
-																	Optional: true,
-																	MinItems: 1,
-																	MaxItems: 8,
-																	Elem: &schema.Schema{
-																		Type:         schema.TypeString,
-																		ValidateFunc: validation.StringLenBetween(1, 255),
-																	},
-																},
-																"includes": {
-																	Type:     schema.TypeList,
-																	Optional: true,
-																	MinItems: 1,
-																	MaxItems: 8,
-																	Elem: &schema.Schema{
-																		Type:         schema.TypeString,
-																		ValidateFunc: validation.StringLenBetween(1, 255),
-																	},
-																},
-															},
-														},
-													},
-												},
-											},
-										},
-										"source_action_name": {
-											Type:     schema.TypeString,
-											Required: true,
-											ValidateFunc: validation.All(
-												validation.StringLenBetween(1, 100),
-												validation.StringMatch(regexache.MustCompile(`[0-9A-Za-z_.@-]+`), ""),
-											),
-										},
-									},
-								},
-							},
-							"provider_type": {
-								Type:             schema.TypeString,
-								Required:         true,
-								ValidateDiagFunc: enum.Validate[types.PipelineTriggerProviderType](),
-							},
-						},
-					},
-				},
+				"trigger":         triggerSchema(),
+				"trigger_all":     sdkv2.DataSourcePropertyFromResourceProperty(triggerSchema()),
```

Comment:
We should probably rename `DataSourcePropertyFromResourceProperty` to capture the fact that it can be used outside of data sources to make al;l attributes _Computed_ only.

Reply:
Agreed. I'll open a tech debt issue for follow up.